portal/src/main/webapp/js/src/oncoprint/webgl/src/js/oncoprintmodel.js

```diff
@@ -818,21 +834,17 @@ var OncoprintModel = (function () {
 	delete this.track_expand_button_getter[track_id];
 	delete this.track_expansion_tracks[track_id];
 
-	var containing_track_group = _getContainingTrackGroup(this, track_id, true);
+	var containing_track_group = _getMajorTrackGroup(this, track_id);
 	if (containing_track_group !== null) {
 	    containing_track_group.splice(
 		    containing_track_group.indexOf(track_id), 1);
 	}
-	// remove any listing of the track as the expansion of another track
-	var group_track, index_in_group;
-	for (group_track in this.track_expansion_tracks) {
-	    if (this.track_expansion_tracks.hasOwnProperty(group_track)) {
-		index_in_group = this.track_expansion_tracks[group_track].indexOf(track_id);
-		if (index_in_group !== -1) {
-		    this.track_expansion_tracks[group_track].splice(index_in_group, 1);
-		}
-	    }
+	// remove listing of the track as an expansion of its parent track
+	var expansion_group = this.track_expansion_tracks[this.track_expansion_parent[track_id]]
+	if (expansion_group) {
+	    expansion_group.splice(expansion_group.indexOf(track_id), 1);
 	}
+	delete this.track_expansion_parent[track_id];
```

Comment:
this always exists?

Reply:
The map itself does, the property for this track ID does not. The same holds for the four other expansion-related track metadata which are deleted a few lines up. The only reason that this one is separated from those by a few lines, is that this line needs the value before it's deleted:
```js
var expansion_group = this.track_expansion_tracks[this.track_expansion_parent[track_id]]
````